Scala/Scala.sublime-syntax

```diff
@@ -85,7 +85,7 @@ contexts:
     - match: |-
         (?x)
         \b(def)\s+
-        (([a-zA-Z$_][a-zA-Z0-9$_]*(_[^a-zA-Z0-9\s]+)?)|`.*`|[^\w\[\(\:\_\s]+)
+        (_*([a-zA-Z$][a-zA-Z0-9$]*|[^a-zA-Z0-9$`\s\(\)\[\]\{\};\.]+)(_+([a-zA-Z$][a-zA-Z0-9$]*|[^a-zA-Z0-9$`\s\(\)\[\]\{\};\._]+)?)*|`[^`]+`|[^\w\[\(\:\_\s]+)
```

Comment:
This regex is in extended mode anyway, does it make sense to expand this a bit? Fully expanded, it looks like this:

```
(
	_*
	(
		[a-zA-Z$][a-zA-Z0-9$]*
	|
		[^a-zA-Z0-9$`\s\(\)\[\]\{\};\.]+
	)
	(
		_+
		(
			[a-zA-Z$][a-zA-Z0-9$]*
		|
			[^a-zA-Z0-9$`\s\(\)\[\]\{\};\._]+
		)?
	)*
|
	`[^`]+`
|
	[^\w\[\(\:\_\s]+
)
```

(…which seems a bit too verbose, there's probably sensible middle ground.)

Referring to that, the matchers at lines 6 and 13 (starting with `[^a-z`…) differ in that the former permits underscores (although that difference probably won't matter in practice).

Excluding underscores lets you write those classes as 
```
[^\w$`\s\(\)\[\]\{\};\.]
```

With this support for symbolic method names, can we drop the last branch of the match? It allows illegal method names like `]]` or `.;` and I can't think of any symbolic names that aren't handled by the matchers you added.